src/com/moclojer/components/logs.clj

```diff
@@ -1,62 +1,137 @@
-                     ["jetty" "hikari" "pedestal" "migratus"])
-           ;; returning false explicitly so Java interop doesn't
-           ;; bugout for some reason in the future.
-           false))))))
-
-(defn clean-timbre-appenders []
-  (->> (reduce-kv
-        (fn [acc k _]
-          (assoc acc k nil))
-        {} (:appenders timbre/*config*))
-       (assoc nil :appenders)
-       timbre/merge-config!))
-
-(defn setup [level stream env]
-  (clean-timbre-appenders)
-  (let [prod? (= env :prod)
-        ns-filter (when prod?
-                    #{"components.*" "back.api.*"
-                      "yaml-generator.*" "cloud-ops.api.*"})
-        appenders (if prod?
-                    (tas/install)
-                    {:appenders
-                     {:println
-                      (core-appenders/println-appender
-                       {:stream stream})}})]
-    (when prod? (clean-dep-logs))
-    (timbre/merge-config!
-     (merge appenders
-            {:min-level level
-             :ns-filter ns-filter}))))
-
-(defmacro log [level & args]
-  `(timbre/log ~level ~@args))
+   [clojure.core.async.impl.channels ManyToManyChannel]))
+
+(let [log! (t/handler:console nil)]
+  (defn log-console!
+    "Uses telemere's console logger explicitly
+    so other added handlers don't trigger."
+    ([level msg]
+     (log-console! level msg nil nil))
+    ([level msg data]
+     (log-console! level msg data nil))
+    ([level msg data error]
+     (log! {:level level
+            :data data
+            :error error
+            :msg_ msg}))))
+
+(defn ->str-values
+  "Adapts all values (including nested maps' values) from given
+  map `m` to a string.
+
+  This is because OpenSearch only accepts string json values."
+  [v]
+  (cond
+    (map? v) (reduce-kv (fn [m k v] (assoc m k (->str-values v))) {} v)
+    (coll? v) (into (empty v) (map ->str-values v))
+    (string? v) v
+    (nil? v) v
+    :else (str v)))
+
+(defn signal->opensearch-log
+  "Adapts a telemere signal to a pre-defined schema for OpenSearch."
+  [{:keys [thread location] :as signal}]
+  (-> (select-keys signal [:level :ctx :data :msg_ :uid :inst])
+      (merge {"thread/group" (:group thread)
+              "thread/name" (:name thread)
+              "thread/id" (:id thread)
+              "location" (str (:ns location) ":"
+                              (:line location) "x"
+                              (:column location))})
+      (->str-values)
+      (assoc :error (:error signal))
+      (json/write-str)))
+
+(defn build-opensearch-base-req
+  [config]
+  (let [{:keys [username password host port index]} config
+        url (str "https://" host ":" port "/_bulk")]
+    {:method :post
+     :url url
+     :basic-auth [username password]
+     :content-type :json
+     :body (json/write-str {:index {:_index index}})}))
+
+(defn send-opensearch-log-req
+  [base-req log]
+  (try
+    (http-client/request
+     (update base-req :body str \newline log \newline))
```

Comment:
what differs this \newline from just the ascii for newline "\n"?

Reply:
\newline is a character, "\n" is a string that contains the escaped symbol for a newline. Both are indistinct in the end, but since the first one is available by clojure itself, why not use it? :smile: